### ISSUE TITLE
実装: non-playing terminal reward v2 を導入して競りPASS崩壊を再検証

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#230]
Parent: #202
Depends on: #228
Base branch: `feature/issue-202-non-playing-ai`

## 背景

#226 で candidate x1 + frozen x4 + 5-seat rotation に変更し、#228 で Frozen bidding opponent を conservative/passive 50:50 混合へ広げた。

しかし30 iterationsの再実験でも bidding candidate はPASS方向へ崩壊した。

### conservative-only run

`~/napoleon_runs/non-playing-ppo-frozen-v1`

iter30:
- PASS rate: 0.9612
- candidate Napoleon formation rate: 0.153
- all-pass forced d=12: 572
- mean reward: 3.608

### conservative/passive 50:50 mixed run

`~/napoleon_runs/non-playing-ppo-mixed-frozen-v1`

iter30:
- PASS rate: 0.9156
- candidate Napoleon formation rate: 0.229
- all-pass forced d=12: 707
- mean reward: 2.936

Frozen opponent分布を広げても崩壊方向は維持されたため、次はterminal role rewardを変更して再検証する。

## 目的

non-playing RLで使用する terminal role reward を v2 に変更し、Napoleonを引き受けるインセンティブを強める。

宣言枚数を `d` とする。

### 勝利時

- Napoleon: `3 * d`
- Adjutant: `d`
- Citizen: `d`
- Napoleon + Adjutant: `4 * d`

### 敗北時

- Napoleon: `-5`
- Adjutant: `0`
- Citizen: `0`
- Napoleon + Adjutant: `-5`

Napoleon+Adjutant は Napoleon と Adjutant の勝利報酬を加算した `4*d` とする。

## 狙い

現行rewardでは、candidateがPASSして市民側へ回る期待値が高くなりやすく、弱いplaying policy環境ではPASSが合理的に学習されている可能性がある。

Reward v2では Napoleon勝利報酬を大きくし、d=13付近で「Napoleonを取る vs 市民側へ回る」の損益分岐を概ね30%前後まで下げることを狙う。

このIssueでは reward 以外の条件は変更しない。

## 実装方針

- reward version / id を新しいversionへ上げる
- bidding / adjutant / exchange の terminal reward 計算を Reward v2 へ統一する
- manifest / checkpoint / run config / provenanceに新しいreward versionを記録する
- 旧reward v1 runを新rewardで誤resumeしないようfail-closeする
- reward以外の rollout topology / opponent mix / PPO設定は維持する

## 維持する条件

- candidate x1 + frozen x4
- candidate 5-seat rotation
- Frozen bidding opponents: conservative/passive 50:50 per-seat seeded mix
- playing: `ppo-separated-v1000` frozen
- PPO hyperparametersは変更しない
- all-pass forced d=12 のゲームルール自体は変更しない

## テスト

最低限、role別rewardを固定fixtureで確認する。

例:

### d=13
- Napoleon win = 39
- Napoleon lose = -5
- Adjutant win = 13
- Adjutant lose = 0
- Citizen win = 13
- Citizen lose = 0
- Napoleon+Adjutant win = 52
- Napoleon+Adjutant lose = -5

### d=19
- Napoleon win = 57
- Adjutant win = 19
- Citizen win = 19
- Napoleon+Adjutant win = 76

all-pass forced contractでは従来どおり `d=12` を用いて Reward v2 を計算する。

## 30 iteration再実験

実装後、前回mixed-frozen runと同じ条件で新規runを実施する。

```text
iterations             30
games-per-iteration    200 logical seeds
epochs                  4
batch-size              128
learning-rate           1e-4
evaluation-interval     5
evaluation-games        100
seed                    202
```

旧runは上書きしない。

新しいoutput directory例:

`~/napoleon_runs/non-playing-ppo-reward-v2`

## 比較指標

iter 5 / 10 / 15 / 20 / 25 / 30 について最低限以下を比較する。

- candidateBiddingDecisionCount
- passRate
- bidCount
- candidateNapoleonFormationRate
- declarationSuccessRate
- allPassForcedD12Count
- meanReward
- targetPointCardsDistribution 13〜19
- fallbackCount
- illegalActionCount

主比較対象は `~/napoleon_runs/non-playing-ppo-mixed-frozen-v1`。

## Acceptance Criteria

- Reward v2 が上記定義で実装される
- Napoleon+Adjutant win が `4*d` になる
- reward version/idが更新され、artifact provenanceに保存される
- reward v1 runをreward v2で誤resumeしない
- candidate x1 + frozen x4 + rotation + mixed frozen bidding topologyを維持する
- fixed playing policyを維持する
- role別rewardのunit testが通る
- 2〜3 iteration smokeが通る
- fallback=0 / illegal action=0
- 実装完了時に30 iteration比較実験用のコピペ可能なコマンドを提示する

## 非目標

- PPOアルゴリズム変更
- learning rate / clip / batch size等のhyperparameter tuning
- Frozen opponent mixの変更
- all-pass forced d=12ルールの変更
- playing policy再学習
- 親PR #203 のmainへのMerge

## PR運用

現在の `napoleon-web-issue-202` 作業ディレクトリをそのまま使用し、新しいworktreeは作らない。

`feature/issue-202-non-playing-ai` を親としてIssue専用ブランチを作成する。
PR baseは `feature/issue-202-non-playing-ai` とする。

Codex review通過・必要テスト通過後は Ready for review にし、そのまま親ブランチへmergeする。`main` にはmergeしない。
